plugin/kubernetes/kubernetes.go

```diff
@@ -244,6 +245,17 @@ func (k *Kubernetes) InitKubeCache(ctx context.Context) (err error) {
 
 	k.opts.zones = k.Zones
 	k.opts.endpointNameMode = k.endpointNameMode
+	// Enable use of endpoint slices if the API supports the discovery v1 beta1 api
+	if _, err := kubeClient.Discovery().ServerResourcesForGroupVersion(discovery.SchemeGroupVersion.String()); err == nil {
+		k.opts.useEndpointSlices = true
```

Comment:
log.Infof this?

Reply:
is this done? I want to resolve my remaining comments, so I can lgtm this

Reply:
Yes. A message is logged when disabling endpointslice watch for k8s before 1.19.